app/src/main/java/com/arctouch/codechallenge/paging/SearchPageKeyedDataSource.kt

```diff
@@ -0,0 +1,144 @@
+package com.arctouch.codechallenge.paging
+
+import android.arch.lifecycle.MutableLiveData
+import android.arch.paging.PageKeyedDataSource
+import com.arctouch.codechallenge.api.ApiService
+import com.arctouch.codechallenge.api.callbacks.TmdbApi
+import com.arctouch.codechallenge.data.Cache
+import com.arctouch.codechallenge.model.Movie
+import com.arctouch.codechallenge.model.UpcomingMoviesResponse
+import com.arctouch.codechallenge.util.*
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+
+class SearchPageKeyedDataSource(var resourceLiveDataSource: MutableLiveData<Resource>, private val query: String) : PageKeyedDataSource<Long, Movie>() {
+
+    override fun loadInitial(params: LoadInitialParams<Long>, callback: LoadInitialCallback<Long, Movie>) {
+        resourceLiveDataSource.postValue(Resource.loading())
+
+        ApiService.getTmdbApiClient().create(TmdbApi::class.java).searchMovies(
+                API_KEY,
+                query,
+                FIRST_PAGE
+        ).enqueue(object : Callback<UpcomingMoviesResponse> {
```

Comment:
Why you chose to use enqueue and callback instead of the Rx that was in the initial version?

Reply:
To be honest, I'm comfortable with the reactive programming paradigm, I can even give you a brief explanation if you need it, the MVVM itself is a reactive programming model. But I've been using a lot of coroutines, and I have not used Rx for a long time, and since it was a two-day challenge, I did not want to risk not doing everything, looking over all the optimal Rx structure again